pkgs/tools/misc/hdf4/default.nix

```diff
@@ -1,99 +1,114 @@
-      sha256 = "0ip4prcjpa404clm87ib7l71605mws54x9492n9pbz5yb51r9aqh";
+      hash = "sha256-EKuUQ1m+/HWTFYmkTormtQATDj0rHlQpI4CoK1m+5EY=";
     })
     (fetchpatch {
       url = "https://src.fedoraproject.org/rpms/hdf/raw/edbe5f49646b609f5bc9aeeee5a2be47e9556e8c/f/hdf-s390.patch";
-      sha256 = "0aiqbr4s1l19y3r3y4wjd5fkv9cfc8rlr4apbh1p0d57wyvqa7i3";
+      hash = "sha256-Ix6Ft+enNHADXFeRTDNijqU9XWmSEz/y8CnQoEleOCo=";
     })
     (fetchpatch {
       url = "https://src.fedoraproject.org/rpms/hdf/raw/edbe5f49646b609f5bc9aeeee5a2be47e9556e8c/f/hdf-arm.patch";
-      sha256 = "157k1avvkpf3x89m1fv4a1kgab6k3jv74rskazrmjivgzav4qaw3";
+      hash = "sha256-gytMtvpvR1nzV1NncrYc0yz1ZlBku1AT6sPdubcK85Q=";
     })
     (fetchpatch {
       url = "https://src.fedoraproject.org/rpms/hdf/raw/edbe5f49646b609f5bc9aeeee5a2be47e9556e8c/f/hdf-aarch64.patch";
-      sha256 = "112svcsilk16ybbsi8ywnxfl2p1v44zh3rfn4ijnl8z08vfqrvvs";
+      hash = "sha256-eu+M3UbgI2plJNblAT8hO1xBXbfco6jX8iZMGjXbWoQ=";
     })
     ./darwin-aarch64.patch
   ];
 
-  nativeBuildInputs = [
-    cmake
-  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    fixDarwinDylibNames
-  ] ++ lib.optional fortranSupport gfortran;
+  nativeBuildInputs =
+    [
+      cmake
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      fixDarwinDylibNames
+    ]
+    ++ lib.optional fortranSupport gfortran;
 
-  buildInputs = [
-    libjpeg
-    zlib
-  ]
-  ++ lib.optional javaSupport jdk
-  ++ lib.optional szipSupport szip
-  ++ lib.optional uselibtirpc libtirpc;
+  buildInputs =
+    [
+      libjpeg
+      zlib
+    ]
+    ++ lib.optional javaSupport jdk
+    ++ lib.optional szipSupport szip
+    ++ lib.optional uselibtirpc libtirpc;
 
-  preConfigure = lib.optionalString uselibtirpc ''
-    # Make tirpc discovery work with CMAKE_PREFIX_PATH
-    substituteInPlace config/cmake/FindXDR.cmake \
-      --replace 'find_path(XDR_INCLUDE_DIR NAMES rpc/types.h PATHS "/usr/include" "/usr/include/tirpc")' \
-                'find_path(XDR_INCLUDE_DIR NAMES rpc/types.h PATH_SUFFIXES include/tirpc)'
-  '' + lib.optionalString szipSupport ''
-    export SZIP_INSTALL=${szip}
-  '';
+  preConfigure =
+    lib.optionalString uselibtirpc ''
+      # Make tirpc discovery work with CMAKE_PREFIX_PATH
+      substituteInPlace config/cmake/FindXDR.cmake \
+        --replace-fail 'find_path(XDR_INCLUDE_DIR NAMES rpc/types.h PATHS "/usr/include" "/usr/include/tirpc")' \
+                       'find_path(XDR_INCLUDE_DIR NAMES rpc/types.h PATH_SUFFIXES include/tirpc)'
+    ''
+    + lib.optionalString szipSupport ''
+      export SZIP_INSTALL=${szip}
+    '';
 
-  cmakeFlags = [
-    "-DBUILD_SHARED_LIBS=ON"
-    "-DHDF4_BUILD_TOOLS=ON"
-    "-DHDF4_BUILD_UTILS=ON"
-    "-DHDF4_BUILD_WITH_INSTALL_NAME=OFF"
-    "-DHDF4_ENABLE_JPEG_LIB_SUPPORT=ON"
-    "-DHDF4_ENABLE_NETCDF=${if netcdfSupport then "ON" else "OFF"}"
-    "-DHDF4_ENABLE_Z_LIB_SUPPORT=ON"
-    "-DJPEG_DIR=${libjpeg}"
```

Comment:
It turns out `JPEG_DIR` no longer does anything, so I removed it. Maybe the library detection got better.